codec-multipart/src/main/java/io/netty/contrib/handler/codec/http/multipart/HttpPostStandardRequestDecoder.java

```diff
@@ -539,11 +539,16 @@ private void parseBodyAttributes() {
 
     private void setFinalBuffer(Buffer buffer) throws IOException {
         currentAttribute.addContent(buffer, true);
-        Buffer decodedBuf = decodeAttribute(currentAttribute.getBuffer(), charset);
+        Buffer currentBuf = currentAttribute.getBuffer();
+        Buffer decodedBuf = decodeAttribute(currentBuf, charset);
         if (decodedBuf != null) { // override content only when ByteBuf needed decoding
             currentAttribute.setContent(decodedBuf);
         }
         addHttpData(currentAttribute);
+        if (! currentAttribute.isInMemory() && currentBuf != null && currentBuf.isAccessible()) {
+            // disked based attribute always returned an allocated buffer from getBuffer method, so we need to close
+            currentBuf.close();
```

Comment:
You may want to investigate if this difference between disk and memory attributes can be eliminated, because that's going to be confusing to people.

Reply:
Indeed, buffer ownership is confusing. I'll try to investigate how to improve this.